app/src/colorpalettewidget.cpp

```diff
@@ -262,6 +264,54 @@ void ColorPaletteWidget::onItemChanged(QListWidgetItem* item)
     editor()->object()->renameColour(index, newColorName);
 }
 
+void ColorPaletteWidget::onRowsMoved(const QModelIndex &parent, int start, int end, const QModelIndex &destination, int row)
+{
+    Q_UNUSED(parent)
+    Q_UNUSED(destination)
+    Q_UNUSED(end)
+
+    int startIndex, endIndex;
+    if (start < row)
+    {
+        row -= 1; // TODO: Is this a bug?
```

Comment:
Not sure... all documentation points towards row and column being zero based (at least for QModelIndex) so in theory when selecting the first item, the row should be 0 but the actual result suggests that it starts at 1.

row in this context might not be the same as row in QModelIndex context but there seems to be no info that should suggests otherwise.

Reply:
I just know that that the default palette has 24 colors, with indexes 0-23. If I drag the first color, and places it in the end, then the 'row' parameter is 24(?). It works if i deduct 1 from row here...

Reply:
Yeah I think it's fine, seems to work as expected from what I've tested.